Migrate welcome message to room description for report

esh-g opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/32615
PROPOSAL: https://github.com/Expensify/App/issues/32615#issuecomment-1870782450


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Create a new workspace.
2. Go to '+' -> 'start chat'
3. Create a new workspace room with a custom description.
4. Make sure the description is shown in the header as well as in the Central pane above composer.
5. Click on the header
6. Click on description
7. Edit the description and try adding markdown
8. Make sure the description is changed in header and above composer
9. Make sure markdown is not rendered in Header.
[The markdown is still not supported on backend (as of 31/Jan)]

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
1. Go offline.
2. Create a new workspace room with a description
3. Make sure the description is shown in chat mast head and header.
4. Click on header and edit the description.
5. Make sure the description menu item gets greyed out after editing offline.
6. Make sure markdown is rendered correctly.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
10. Upload an image via copy paste
11. Verify a modal appears displaying a preview of that image
--->

1. Create a new workspace.
2. Go to '+' -> 'start chat'
3. Create a new workspace room with a custom description.
4. Make sure the description is shown in the header as well as in the Central pane above composer.
5. Click on the header
6. Click on description
7. Edit the description and try adding markdown
8. Make sure the description is changed in header and above composer
9. Make sure markdown is not rendered in Header.
[The markdown is still not supported on backend (as of 31/Jan)]


- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->


https://github.com/Expensify/App/assets/77237602/158af4d9-ce82-42b6-96bc-be78ff136462



</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/77237602/dcc4becb-de91-4929-a192-1631093cd317


</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/77237602/128b8256-1044-4de3-937a-274222fe46be


</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/77237602/2395ecf7-bc07-48b8-9ad8-bedc1d17f4af



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/77237602/c9e3d9e4-c697-457d-bd80-6f6771468994


https://github.com/Expensify/App/assets/77237602/930c27c4-5600-4759-8b3e-2e198bdad81d


</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/77237602/a84585e9-ae4b-4089-9311-7d3a119f60dd


</details>

esh-g committed:
Migrate welcome message to room description for report

esh-g commented:
Some things that are pending:
- Update `AddWorkspaceRoom` to accept `description` instead of `welcomeMessage` on backend.
- Make the Backend capable of saving the `description` as html instead of converting to plain text.
- Make sure the translations and copy added here is as expected. (for room description)
- Look into how to reconcile tasks/description with report/description (currently, I made it `/roomDescription`)

esh-g committed:
Fix style 💫

esh-g committed:
Fix Permissions

puneetlath commented:
>Update AddWorkspaceRoom to accept description instead of welcomeMessage on backend.

This is in progress. I'll let you know when it's live. 

>Make the Backend capable of saving the description as html instead of converting to plain text.

This is in-progress. We'll save it as JSON `{'html': 'this is the description'}' like we do for the welcome message today. 

>Make sure the translations and copy added here is as expected. (for room description)

Can you post these in the #expensify-open-source channel and I'll make sure they get confirmed. 

>Look into how to reconcile tasks/description with report/description (currently, I made it /roomDescription)

Can you expand on this? What is the problem?

esh-g commented:
> Can you post these in the #expensify-open-source channel and I'll make sure they get confirmed.
I have posted the translations [here](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q)

esh-g commented:
gentle bump @rushatgabhane on the review...

rushatgabhane commented on src/pages/ReportDescriptionPage.js:
```diff
@@ -80,39 +80,39 @@ function ReportWelcomeMessagePage(props) {
                 <FormProvider
                     style={[styles.flexGrow1, styles.ph5]}
-                    formID={ONYXKEYS.FORMS.WELCOME_MESSAGE_FORM}
+                    formID={ONYXKEYS.FORMS.REPORT_DESCRIPTION}
```
Bug: 
1. console error on opening room description page.
2. setting a room description and navigating to room description crashes the app

https://github.com/Expensify/App/assets/29673073/24731d74-fd11-43e5-99a6-0109b8193add

esh-g replied:
@rushatgabhane As for the first bug, I just noticed it was because of a typo in the `_FORM` at the end. But I wasn't able to reproduce the crash. Reading the stack trace from the crash, it doesn't seem to be related to the code edited... Could you please confirm if it is reproducible reliably?

https://github.com/Expensify/App/assets/77237602/3e3e8a91-e910-4d27-88af-03b51caa1e05

rushatgabhane commented on src/pages/ReportDescriptionPage.js:
```diff
@@ -80,39 +80,39 @@ function ReportWelcomeMessagePage(props) {
                                 }
-                                welcomeMessageInputRef.current = el;
-                                updateMultilineInputRange(welcomeMessageInputRef.current);
+                                reportDescriptionInputRef.current = el;
```
Bug: navigate to an existing workspace room that doesn't have description.
It shows not found page.

https://github.com/Expensify/App/assets/29673073/2e7d3f30-225a-400e-aa77-ca251407d1b8

esh-g replied:
@rushatgabhane Currently we are showing the `notFound` page for those reports which are being excluded in this function:
<img width="1134" alt="Screenshot 2024-01-14 at 10 10 45 PM" src="https://github.com/Expensify/App/assets/77237602/c11fb308-1662-4900-a103-eabf1655a891">

But I agree we need to change this. I'll make some modifications now

esh-g replied:
Since this is a policy expense chat, there shouldn't be any room description. We are working here: https://github.com/Expensify/App/issues/32955#issue-2038816407 to show only the policy description for the policy expense chat.

esh-g replied:
Here: https://github.com/Expensify/App/pull/34150/commits/e28f6cddaf021ce341d75b4a631b1a5a5ffaa3a4
I made the change that description is not visible for a report if:
- The User cannot edit description AND description is empty
But visible if:
- The user can edit OR description is not empty.
This way I don't think there will be a not found page in the flow.

rushatgabhane commented on src/pages/ReportDetailsPage.js:
```diff
@@ -235,6 +236,13 @@ function ReportDetailsPage(props) {
                             )}
                         </View>
                     </View>
+                    <MenuItemWithTopDescription
```
The styles should match the mockups in the issue

<img width="876" alt="image" src="https://github.com/Expensify/App/assets/29673073/b9b69d54-f035-4c22-81b4-19d2d8c28f46">

esh-g replied:
As for styles do you mean the text `Description` vs the `Room Description` text in the mockup? Or is there something else as well?

esh-g replied:
Made it room description here: https://github.com/Expensify/App/pull/34150/commits/6e7fc2761fac092106366d45e54e92a9fc10fb63

esh-g committed:
Requested changes

esh-g committed:
Hide Description conditionally

rushatgabhane commented:
@esh-g thank you for the changes. The app doesn't crash now

rushatgabhane commented:
Bug: Can't set room description when creating a room inside a workspace that the user doesn't own. They are a member tho.

<img width="1076" alt="image" src="https://github.com/Expensify/App/assets/29673073/6a31139b-cbd7-4012-9b55-0434477455ee">

rushatgabhane commented:
Bug: when creating a new room with a descpription, the description doesn't persist.
1. New room -> set description when creating the room
Expected: room description persists.
Actual: description disappears

https://github.com/Expensify/App/assets/29673073/74c8c112-9ee0-414e-a519-0af7acf426ed

puneetlath commented:
>Update AddWorkspaceRoom to accept description instead of welcomeMessage on backend.

This is live now on staging. 

>Make the Backend capable of saving the description as html instead of converting to plain text.

This is live now on staging. 

>Make sure the translations and copy added here is as expected. (for room description)

Asked in the internal channel. Should hear back on this soon. 

>@puneetlath Currently, the path r/:reportID/description is used only for setting task descriptions. So, what I did was that I set the report description path to r/:reportID/roomDescription as it was only specific to a room. But I'm not sure if we mean to reconcile these routes somehow or rename them to /description and /taskDescription or maybe just use /description for both.

I think we could use /description for both. Unless it creates some sort of problem. Ultimately, they are both for setting the "description" field on a report.

esh-g committed:
Update spanish translations

esh-g commented:
@puneetlath @rushatgabhane I have another question:
When a workspace member (who isn't an admin) creates a new room, they are able to set the description on creation but are not able to edit it since we currently only allow admins. Is this something that should be changed?

esh-g committed:
Merge branch 'main' into room-description

esh-g commented:
> I think we could use /description for both. Unless it creates some sort of problem. Ultimately, they are both for setting the "description" field on a report.

To do that, I'm thinking of making the `/description` correspond to `ReportDescriptionPage` and then check if it is a task:
- If task: Render the `TaskDescription` page
- else: Render `RoomDescriptionPage`(new)

esh-g committed:
Fix header for tasks

puneetlath commented:
>When a workspace member (who isn't an admin) creates a new room, they are able to set the description on creation but are not able to edit it since we currently only allow admins. Is this something that should be changed?

Yes, any workspace member who is also a member of the room should be able to change the description. 

>To do that, I'm thinking of making the /description correspond to ReportDescriptionPage and then check if it is a task:

Makes sense to me. cc @rushatgabhane in case you have any concerns.

rushatgabhane commented:
> To do that, I'm thinking of making the /description correspond to ReportDescriptionPage and then check if it is a task:

> Makes sense to me. cc @rushatgabhane in case you have any concerns.

To achieve same route paths `/description`, we would be pushing the logic from router to code.



Also, from product perspective, the task description is same as a room description. i.e. a short summary of what the task is / what the room is for.

I'm okay with this.

esh-g committed:
Reconcile description route and permission

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix lint

esh-g committed:
Fix TS

esh-g commented:
@puneetlath @rushatgabhane I made all the changes with the `/description` and the permission to edit for all policy members.

esh-g committed:
Fix permission

esh-g commented:
The TS is failing because I moved the code from `reportDescription` to `roomDescription`. I would have migrated but components like `FormProvider` and `withReportOrNotFound()` are still not migrated and causing errors. Hope we can merge without that.

esh-g commented:
gentle bump @rushatgabhane

esh-g committed:
Merge branch 'main' into room-description

quinthar commented:
@rushatgabhane what's your ETA for review?

puneetlath commented:
Requested a new C+ here: https://expensify.slack.com/archives/C02NK2DQWUX/p1706195245170869. @ishpaul777 is going to take over 

Sorry for the delay @esh-g

quinthar commented:
Great, what's the next step, who's doing it, and when?

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix conflict

esh-g committed:
Migrate TS

esh-g commented:
@ishpaul777 Made all the requested changes. Please test once by going offline and make sure the markdown stuff works as well

esh-g committed:
Fix TS

esh-g commented:
@puneetlath I have a question:
Currently when editing report description offline, we don't have any indication of `OfflineWithFeedback` greying out. Is this as expected or should the `OfflineWithFeedback` be added. If yes, could that be done in a follow-up issue?

https://github.com/Expensify/App/assets/77237602/0c445c30-742b-48c8-b0d0-05f422c9e30c

esh-g commented on src/components/ReportWelcomeText.tsx:
```diff
@@ -69,17 +70,23 @@ function ReportWelcomeText({report, policy, personalDetails}: ReportWelcomeTextP
-                                {ReportUtils.getReportName(report)}
-                            </Text>
+                        {report?.description ? (
+                            <RenderHTML html={report.description} />
```
seems like the issue was edited after I got assigned... I'll implement the edited changes as well

esh-g replied:
Done here: https://github.com/Expensify/App/pull/34150/commits/9b4d4819b335002186b1a1f8ab1eefcc423201e9

esh-g commented:
<img width="750" alt="Screenshot 2024-01-26 at 8 45 54 PM" src="https://github.com/Expensify/App/assets/77237602/5d7dd14b-37d5-449b-8f8e-97b699207f1b">
@puneetlath Could we please have a design for this? It is not possible to truncate with the `numberOfLines` prop because we are rendering html and not using `Text`. So, should we set a specific height instead? And moreover, should the options below the description like `share code`, `members`, `settings`, etc be hidden with the `show more` button as well?

esh-g committed:
Pressable redirect

esh-g committed:
Fix

esh-g committed:
Make permission changes

esh-g commented on src/pages/home/HeaderView.js:
```diff
@@ -272,29 +274,56 @@ function HeaderView(props) {
+                                                shouldUseFullTitle={isChatRoom || isPolicyExpenseChat || isChatThread || isTaskReport}
+                                            />
+                                            {!_.isEmpty(policyName) &&
+                                                !_.isEmpty(reportDescription) && ( // code for the policyName display
```
<img width="419" alt="Screenshot 2024-01-26 at 9 44 48 PM" src="https://github.com/Expensify/App/assets/77237602/7bbc3f57-ccda-4458-893e-f614d6dd4373">
Currently we are showing workspace name as subtitle if description doesn't exist. I think this looks good...

quinthar commented:
For this are you suggesting:

1. If there IS a workspace description, put the `<room> in <workspaceName>` in the first line, and `<workspaceDescription>` in the second
2. If there IS NOT a workspace description, put `<room>` in the first line, and `<workspaceName>` in the second

if so, that sounds great!

<img width="792" alt="image" src="https://github.com/Expensify/App/assets/575186/8964b7df-8781-44c4-a035-88ede4f6e0b2">

quinthar commented:
> Could we please have a design for this? It is not possible to truncate with the numberOfLines prop because we are rendering html and not using Text. So, should we set a specific height instead?

Hm, ok for now just show the whole room description, and just limit the description input to like 1024 characters. 

> And moreover, should the options below the description like share code, members, settings, etc be hidden with the show more button as well?

No, the "show more" was only there to expand the truncated description.  But if we aren't truncating the description, then let's not worry about that.  So always show the other buttons.  Does this make sense?

esh-g commented:
@quinthar I have one final question here: https://github.com/Expensify/App/pull/34150#issuecomment-1911564404 about the offline behaviour of editing the desceiption

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix conflict

esh-g committed:
Limit characters to 1024

esh-g commented:
I limited the characters and I think all the requirements are completed. @ishpaul777 Could you please check if anything else is pending? As for the `offline` behaviour question, I think that can be handled in a follow-up PR issue if needed, as this PR is already very big and I hope we can merge this before any more conflicts occur...

esh-g committed:
Fix lint

esh-g committed:
Improvement of header for small screens

esh-g committed:
Add offline pattern

esh-g committed:
Fix new task description

esh-g committed:
hide description menuItem for tasks

esh-g committed:
rename AdditionalComponent -> renderAdditionalText

esh-g committed:
Fix lint

esh-g commented:
> Not found page on clicking description on when creating a task.
Fixed here: https://github.com/Expensify/App/pull/34150/commits/c2ad5db8ab4eeff30cb0569d68dd905138ad0864

> When opening task details, It says Room description while it is task description and also it is not editable.
Fixed here: https://github.com/Expensify/App/pull/34150/commits/52db819f43e7e1ecc5eda1a574d9d52d7ae20849

> can we call this prop renderAdditionalText ?
Done here: https://github.com/Expensify/App/pull/34150/commits/0dffd611153236052abf098521b8bc8c0c344f98

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix header

esh-g committed:
Make in lowercase

esh-g committed:
Fixed 'in' bold issue

esh-g committed:
Change condition

esh-g committed:
Remove welcome message

esh-g commented:
@ishpaul777 Sorry I'm having trouble with pods which I have been trying to fix from the last two days. Could you please help me with that test?

esh-g commented:
@ishpaul777 Rectified the test steps and added note

esh-g commented:
@ishpaul777 I'm not able to reproduce the issue on android.. I have the video attached as it seems for me. Are there any testing steps?

esh-g committed:
Fix android

esh-g commented:
@ishpaul777 I modified the code to use text spacing instead of margins within text... maybe this will fair better for native platforms. Could you please test again?

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix chat mast head text

esh-g committed:
Fix lint

esh-g committed:
Fix tests

esh-g committed:
Fix lint

esh-g committed:
Limit length

esh-g committed:
Fix lint

esh-g committed:
wrap html

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix conflicts

esh-g committed:
Disable enter prop

esh-g committed:
Fix lint

esh-g commented:
Gentle bump @ishpaul777

esh-g commented:
<img width="916" alt="Screenshot 2024-02-04 at 2 00 10 PM" src="https://github.com/Expensify/App/assets/77237602/b0739723-a2f7-44ee-b031-9fe6890936cb">

@ishpaul777 I did respond before and even asked if I should resolve that... Seems like GH glitched somehow

esh-g commented:
@ishpaul777 Yes we can, but I figured it doesn't make sense to have the code check for anonymous action if that action is disabled anyway

esh-g committed:
Anonymous action check

esh-g commented:
@ishpaul777 Although I don't see how this would cause a regression, but since we probably don't need the checking for anonymous action of the description as it would hopefully be disabled for anonymous user, so we can go ahead with that

ishpaul777 reviewed (approved):
Apart from Backend not handling the markdown properly and its side effect issue https://github.com/Expensify/App/pull/34150#issuecomment-1921198062, rest looks good and tests well! 🚀

esh-g committed:
Merge branch 'main' into room-description

esh-g committed:
Fix conflict

esh-g commented:
@ishpaul777 Fixed the conflict.
@puneetlath Hope we can try to merge this before any conflict arises again (had to fix like 7-8 conflicts for this 😅)
Thanks!

esh-g committed:
Prettify

puneetlath commented on src/libs/E2E/apiMocks/openReport.ts:
```diff
@@ -0,0 +1,1971 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Can you help me understand what this file does?

esh-g committed:
Remove openReport

puneetlath commented:
Is it a bug that the room description isn't in this menu? Or was that not part of the scope? 

<img width="384" alt="image" src="https://github.com/Expensify/App/assets/1371996/1d20ea91-13bc-48ab-8594-7e431f149b98">

puneetlath commented:
It seems like it wasn't part of the scope, though I think it should be. We can handle that as a follow-up.

puneetlath commented on src/libs/ReportUtils.ts:
```diff
@@ -4585,10 +4594,17 @@ function getRoom(type: ValueOf<typeof CONST.REPORT.CHAT_TYPE>, policyID: string)
 
 /**
- *  We only want policy owners and admins to be able to modify the welcome message, but not in thread chat.
+ *  We only want policy owners and admins to be able to modify the report description, but not in thread chat.
```
```suggestion
 *  We only want policy members who are members of the report to be able to modify the report description, but not in thread chat.
```

puneetlath commented on src/pages/RoomDescriptionPage.tsx:
```diff
@@ -0,0 +1,113 @@
+    const policy = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`];
+
+    const handleReportDescriptionChange = useCallback((value: string) => {
+        setdescription(value);
```
```suggestion
        setDescription(value);
```

puneetlath commented on src/pages/RoomDescriptionPage.tsx:
```diff
@@ -0,0 +1,113 @@
+function RoomDescriptionPage({report, policies}: RoomDescriptionPageProps) {
+    const styles = useThemeStyles();
+    const parser = new ExpensiMark();
+    const [description, setdescription] = useState(() => parser.htmlToMarkdown(report?.description ?? ''));
```
```suggestion
    const [description, setDescription] = useState(() => parser.htmlToMarkdown(report?.description ?? ''));
```

puneetlath reviewed (changes requested):
Some very minor comments, great job with this!

esh-g committed:
Update src/libs/ReportUtils.ts

Co-authored-by: Puneet Lath <puneet@expensify.com>

esh-g committed:
Update src/pages/RoomDescriptionPage.tsx

Co-authored-by: Puneet Lath <puneet@expensify.com>

esh-g committed:
Update src/pages/RoomDescriptionPage.tsx

Co-authored-by: Puneet Lath <puneet@expensify.com>

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/7791082209) to staging by https://github.com/thienlnam in version: 1.4.36-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/7791082209) to staging by https://github.com/thienlnam in version: 1.4.36-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

lanitochka17 commented:
This PR is failing because of issue # https://github.com/Expensify/App/issues/35901

The issue is reproducible in: Web, mWeb/Chrome,  Android, IOS


https://github.com/Expensify/App/assets/78819774/40ea9381-a4a9-4cf7-be04-2984b3cf53c0

esh-g commented:
Yes, this is because the backend markdown handling is pending
cc @puneetlath

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7812774505) to production by https://github.com/Julesssss in version: 1.4.37-7 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

mananjadhav commented:
@ishpaul777 @esh-g We set `disablePressOnEnter={false}` in `WorkspaceNewRoomPage` in [this commit](https://github.com/Expensify/App/pull/34150/commits/5ddfaacd08386ba193927981b90a589bd9663f7a), can we confirm why we explicitly set this? I am currently reviewing this [issue](https://github.com/Expensify/App/issues/36624).

dmkt9 commented on src/pages/home/HeaderView.js:
```diff
@@ -112,6 +114,8 @@ function HeaderView(props) {
     const isPolicyMember = useMemo(() => !_.isEmpty(props.policy), [props.policy]);
     const canLeaveRoom = ReportUtils.canLeaveRoom(props.report, isPolicyMember);
     const isArchivedRoom = ReportUtils.isArchivedRoom(props.report);
+    const reportDescription = ReportUtils.getReportDescriptionText(props.report);
```
Coming from https://github.com/Expensify/App/issues/87093, `reportDescription` is always rendered with `numberOfLines={1}`. However, because Web, Android, and iOS handle multi-line text differently, this results in display inconsistencies across these platforms.
